src/main/java/xyz/thepathfinder/android/Action.java

```diff
@@ -149,16 +154,16 @@ private static double getLongitude(JsonObject json) {
      * @throws IllegalArgumentException when the model isn't a transport or commodity.
      */
     private static SubscribableCrudModel getModel(JsonObject json, PathfinderServices services) {
-        JsonObject model = json.getAsJsonObject("model");
-        String type = model.get("model").getAsString();
+        JsonObject model = json.getAsJsonObject("commodity");
```

Comment:
Maybe I'm not understanding this method, but it it forcing every route action to be a commodity? B/c the name and comments seem to indicate that it could also be a transport, but it looks like the code only uses commodities.